docs/storage/commit_log/main.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian/docs/storage/commit_log/election"
+	"github.com/google/trillian/docs/storage/commit_log/kafka"
+	"github.com/google/trillian/docs/storage/commit_log/signer"
+)
+
+var (
+	signerCount         = flag.Int("signer_count", 3, "Number of parallel signers to run")
+	leafInterval        = flag.Duration("leaf_interval", 500*time.Millisecond, "Period between added leaves")
+	eventInterval       = flag.Duration("event_interval", 1*time.Second, "Interval between events")
+	masterChangePercent = flag.Int("master_change", 20, "Percent chance of a change of master")
+	dualMasterPercent   = flag.Int("dual_master", 8, "Percent chance of a dual master")
+	leafTogglePercent   = flag.Int("leaf_toggle", 10, "Percent chance of toggling leaf generation")
+)
+
+func signerName(i int) string {
+	switch i {
```

Comment:
You could do this in fewer lines with something like this:
```golang
var nums = []string{"one", "two", "three", ..., "nine"}
...
if i >=0 && i < 9 {
return nums[i-1]
} else {
return fmt...
}
```

Or even go all in:
```golang
const nums = "onetwothreefour"
var offs = []struct{ s, e int }{{0, 3}, {3, 6}, {6, 11}, {11, 15}}
...
if i > len(offs) {return fmt.Sprintf("signer%d", i)}
i := func(i int) int {if i >= len(offs) {return len(offs)} else {return i}}(i-1)
return nums[offs[i].s:offs[i].e]
```
(or maybe not :)